kola/tests/misc/selinux.go

```diff
@@ -56,13 +72,90 @@ func SelinuxEnforce(c cluster.TestCluster) {
 		c.Fatalf("failed to reboot machine: %v", err)
 	}
+}
+
+// SelinuxEnforce checks that some basic things work after `setenforce 1`
+func SelinuxEnforce(c cluster.TestCluster) {
+	cmdList := []cmdCheckOutput{
+		{"/usr/sbin/getenforce", true, "Permissive"},
+		{"sudo /usr/sbin/setenforce 0", false, ""},
+		{"/usr/sbin/getenforce", true, "Permissive"},
+		{"sudo /usr/sbin/setenforce 1", false, ""},
+		{"/usr/sbin/getenforce", true, "Enforcing"},
+		{"systemctl --no-pager is-active system.slice", true, "active"},
+		{"sudo cp /etc/selinux/config{,.old}", false, ""},
+		{"sudo sed -i 's/SELINUX=permissive/SELINUX=enforcing/' /etc/selinux/config", false, ""},
+	}
+
+	m := c.Machines()[0]
 
-	o := c.MustSSH(m, "which getenforce")
-	c.Logf(string(o))
+	testSelinuxCmds(c, m, cmdList)
 
-	output := c.MustSSH(m, "getenforce")
+	output := c.MustSSH(m, "/usr/sbin/getenforce")
 
 	if string(output) != "Enforcing" {
-		c.Fatalf("command \"getenforce\" has unexpected output: want \"Enforcing\" got %q", string(output))
+		c.Fatalf("command \"/usr/sbin/getenforce\" has unexpected output: want \"Enforcing\", got %q", string(output))
+	}
+}
+
+// SelinuxBoolean checks that you can tweak a boolean in the current session
+func SelinuxBoolean(c cluster.TestCluster) {
+	cmdList := []cmdCheckOutput{
+		{"/usr/sbin/getsebool virt_use_nfs", true, "virt_use_nfs --> off"},
+		{"sudo /usr/sbin/setsebool virt_use_nfs 1", false, ""},
+		{"/usr/sbin/getsebool virt_use_nfs", true, "virt_use_nfs --> on"},
+	}
+
+	m := c.Machines()[0]
+
+	testSelinuxCmds(c, m, cmdList)
+
+	// since we didn't persist the change, should return to default value
+	output := c.MustSSH(m, "/usr/sbin/getsebool virt_use_nfs")
+
+	if string(output) != "virt_use_nfs --> off" {
+		c.Fatalf("The SELinux boolean \"virt_use_nfs\" is incorrectly configured: want \"virt_use_nfs --> off\", got %q", string(output))
+	}
+}
+
+// SelinuxBooleanPersist checks that you can tweak a boolean and have it
+// persist across reboots
+func SelinuxBooleanPersist(c cluster.TestCluster) {
+	cmdList := []cmdCheckOutput{
+		{"/usr/sbin/getsebool virt_use_nfs", true, "virt_use_nfs --> off"},
+		{"sudo /usr/sbin/setsebool -P virt_use_nfs 1", false, ""},
+		{"/usr/sbin/getsebool virt_use_nfs", true, "virt_use_nfs --> on"},
+	}
+
+	m := c.Machines()[0]
+
+	testSelinuxCmds(c, m, cmdList)
+
+	// the change should be persisted after a reboot
+	output := c.MustSSH(m, "/usr/sbin/getsebool virt_use_nfs")
+
+	if string(output) != "virt_use_nfs --> on" {
+		c.Fatalf("The SELinux boolean \"virt_use_nfs\" is incorrectly configured: want \"virt_use_nfs --> on\", got %q", string(output))
+	}
+}
+
+// SelinuxManage checks that you can modify an SELinux file context and
+// have it persist across reboots
+func SelinuxManage(c cluster.TestCluster) {
+	cmdList := []cmdCheckOutput{
+		{"sudo /usr/sbin/semanage fcontext -l | grep vasd", true, "/var/opt/quest/vas/vasd(/.*)?                      all files          system_u:object_r:var_auth_t:s0"},
```

Comment:
I'm wondering whether or not it might be worth it to switch the output validation string to a regexp rather than doing a direct string comparison, thoughts?

Reply:
Yeah, that makes sense given how long the output is here.  I'll investigate.